src/bonefish/serialization/json_serializer.cpp

```diff
@@ -167,11 +167,11 @@ struct wamp_bin_string_conversion
 
 wamp_message* json_serializer::deserialize(const char* buffer, size_t length) const
 {
-    msgpack::zone zone;
     msgpack::object item;
+    std::unique_ptr<msgpack::zone> zone(new msgpack::zone());
 
     imemstream bufferstream(buffer, length);
-    serialization::msgpack_from_json_handler<wamp_bin_string_conversion> handler(item, zone);
+    serialization::msgpack_from_json_handler<wamp_bin_string_conversion> handler(item, *(zone.get()));
```

Comment:
`*zone` is a more concise way to ask for a reference to the object in a smart pointer.